### PR TITLE
Add support for Win7 and 32-bit msvc+vcpkg builds

### PR DESCRIPTION
**Pull request by IvanSavenko**
- Prebuilts package for vcpkg is now generated from scratch using Github CI (see https://github.com/vcmi/vcmi-deps-windows/pull/5 )
- As a result, all libraries were updated to latest version available on vcpkg
- Now only boost components that are required for vcmi are included into the package instead of entire 'boost' package (due to issues with boost-python on Win7/32 bit)
- Package now archived as .txz instead of .7z (hopefully will improve extraction times - currently msvc CI spends 7+ minutes to extract package)
- Added / restored support for 32-bit builds. Tested on 64-bit system, but not on native 32 bit one.
- VCPKG now targets Windows 7, but untested.

With this changes we can now completely switch to vcpkg / msvc as default release package for Windows and drop 'alternate' builds. Should also make prebuilts package updates less of a pain since we no longer need to wait for developer with msvc setup to do everything on his PC.

In future this can be replaced by conan, but replacing vcpkg with conan for msvc looks to be non-trivial at the moment, while automating vcpkg is simple enough.

I've also checked possibility of windows-arm builds via vcpkg, but seems to not be possible - qt cross-compilation is broken on vcpkg for years (or never worked to begin with). So not possible until vcpkg fixes that or Github provides arm Windows runners.

### CODE REVIEW

**Comment by Laserlicht** [comment]
> I've also checked possibility of windows-arm builds via vcpkg, but seems to not be possible - qt cross-compilation is broken on vcpkg for years (or never worked to begin with). So not possible until vcpkg fixes that or Github provides arm Windows runners.

Would an ARM64 linux somehow help here? Apparently ARM64 ubuntu runners should be available by the end of the year:
https://github.blog/news-insights/product-news/arm64-on-github-actions-powering-faster-more-efficient-build-systems/

**Comment by IvanSavenko** [comment]
Linux arm runners - no.
But they also mention Windows arm runners which should work in theory.

Our mingw builds use cross-compilation toolchain provided by Ubuntu. And that toolchain only comes in i686 and x64 varieties, so no arm64:
https://packages.ubuntu.com/noble/binutils-mingw-w64-i686
https://packages.ubuntu.com/noble/binutils-mingw-w64-x86-64
I think some distros also have arm64 version of this toolchain, but not Ubuntu. And Github only provides Ubuntu runners for Linux.

So from my point of view windows-arm package at the moment is too low on cost/benefit ratio.

**Comment by kambala-decapitator** [comment]
> qt cross-compilation is broken on vcpkg for years (or never worked to begin with)

Then it's funny how/why x64 -> x86 cross-compiling works. Iirc for msvc it shouldn't really matter which cross-compiler is used (but I never tried the arm one).

BTW that's exactly easy stopped me from using vcpkg for macos / ios in the first place.

**Comment by IvanSavenko** [comment]
> Then it's funny how/why x64 -> x86 cross-compiling works. Iirc for msvc it shouldn't really matter which cross-compiler is used (but I never tried the arm one).

In my attempt it failed because x64 Windows attempted to run arm64 binary (some freshly compiled qt tool I guess?). But x64 Windows can run x86 binary just fine, so x64 -> x86 works.

File: CI/before_install/msvc.sh [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+DUMPBIN_PATH="$MSVC_INSTALL_PATH"/VC/Tools/MSVC/"$TOOLS_DIR"/bin/Hostx64/x64/dumpbin.exe
+
+# This script should also work, but for some reason is *extremely* slow on Github CI (~7 minutes)
+#DUMPBIN_PATH=$(vswhere -latest -find **/dumpbin.exe | head -n 1)
+#
+
+echo "TOOLS_DIR = $TOOLS_DIR"
+echo "DUMPBIN_PATH = $DUMPBIN_PATH"
+
+dirname "$DUMPBIN_PATH" > "$GITHUB_PATH"
```
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply CI/before_install/msvc.sh:18]: Was like this before, so had no reason to change

File: .github/workflows/github.yml [hunk 2]
```diff
@@ -232,7 +244,7 @@ jobs:
         elif [[ (${{matrix.preset}} == android-conan-ninja-release) && (${{github.ref}} != 'refs/heads/master') ]]
         then
             cmake -DENABLE_CCACHE:BOOL=ON -DANDROID_GRADLE_PROPERTIES="applicationIdSuffix=.daily;signingConfig=dailySigning;applicationLabel=VCMI daily" --preset ${{ matrix.preset }}
-        elif [[ ${{matrix.platform}} != msvc ]]
+        elif [[ (${{matrix.platform}} != msvc-x64) && (${{matrix.platform}} != msvc-x86) ]]
```
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply .github/workflows/github.yml:247]: You mean `startsWith(matrix.platform, 'msvc')`? Or equivalent bash expression?
From what I understand, `startsWith` is Github Actions syntax, but this block is shell commands to execute (bash in our case)

**Comment by kambala-decapitator** [comment]
> In my attempt it failed because x64 Windows attempted to run arm64 binary (some freshly compiled qt tool I guess?). But x64 Windows can run x86 binary just fine, so x64 -> x86 works.

Qt 5? It should actually build tools for host, not target. But seems that Qt 5 simply doesn't support cross-compiling on Windows  or msvc (dunno about 6).

File: .github/workflows/github.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kambala-decapitator) [reply .github/workflows/github.yml:247]: Better use GhA syntax because it works basically like a text preprocessor

**Comment by IvanSavenko** [comment]
Yep, one that we use for vcmi: `qt5-base qt5-tools`

In my attempt, VCPKG did built tools for host system (x64), but I still ended up with failure due to "executable format not supported" or something like that. I guess there is a bug somewhere in vcpkg and/or qt5, that leads to attempt to run some binary with target arch instead of host arch.

Haven't tried qt6.

**Pull request merged**